pkg/controller/eventinstance/eventinstance.go

```diff
@@ -0,0 +1,91 @@
+package eventinstance
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	v1 "github.com/acorn-io/acorn/pkg/apis/internal.acorn.io/v1"
+	"github.com/acorn-io/acorn/pkg/config"
+	"github.com/acorn-io/baaah/pkg/router"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	kclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const defaultTTL = 24 * time.Hour
+
+func GCExpired() router.HandlerFunc {
+	// last stores pre-parsed TTLs from the configuration.
+	var last struct {
+		sync.Mutex
+		raw string
+		ttl time.Duration
+	}
+
+	return handler{
+		ttl: func(ctx context.Context, getter kclient.Reader) (time.Duration, error) {
+			cfg, err := config.Get(ctx, getter)
```

Comment:
Would it be better if we can initialize the value so we have to call this every time event is synced? Given the amount of the event it generates I think we will be calling this a lot

Reply:
Yes, I have an idea on this.

What if the value is read when the handler is initialized. Then another handler function is created that watches the config. When the config changes, the handler reads the new TTL value only then.

Thoughts?

Reply:
It is called a bunch, but the value _can_ change so we don't want to set and forget.

I think there's a better approach that involves keeping a shared atomic up to date with a separate config controller/watch/etc. WDYT?

Reply:
@njhale I don't think we need shared memory. I think the `handler` you created can have another `HandlerFunc` on it that watches the config and updates its TTL value.

Reply:
> Or I wonder if we have some mechanism like, for important setting X,Y,Z, if that changes, exit and restarting controller. Then we don't have to worry about dynamically change the values

We could. but that seems like an entirely novel concept for our controllers, and I thought there was already precedent for dynamically adjusting to the latest config in other handlers?

Reply:
> I don't think we need shared memory. I think the handler you created can have another HandlerFunc on it that watches the config and updates its TTL value.

@thedadams If I'm understanding you correctly, this is where I was going with my statement, although I would still consider the `handler.ttl` field a critical region since we can have multiple goroutines reading and writing it.

Reply:
OK -- so I've refactored a bunch of this so that we're no longer invoking the getter on every event reconciliation. PTAL @thedadams @StrongMonkey and LMK if it makes sense to you.

Reply:
scratch that, rolled back those changes.